flink-dist/src/test/java/org/apache/flink/dist/BashJavaUtilsITCase.java

```diff
@@ -34,21 +36,30 @@
 
 	private static final String RUN_BASH_JAVA_UTILS_CMD_SCRIPT = "src/test/bin/runBashJavaUtilsCmd.sh";
 
+	/**
+	 * Executes the given shell script wrapper and returns the last line.
+	 */
+	private String executeScriptAndFetchLastLine(final String command) throws IOException {
+		String[] commands = {RUN_BASH_JAVA_UTILS_CMD_SCRIPT, command};
+		String[] lines = executeScript(commands).split(System.lineSeparator());
+		if (lines.length == 0) {
+			return "";
+		} else {
+			return lines[lines.length - 1];
+		}
+	}
+
 	@Test
 	public void testGetTmResourceDynamicConfigs() throws Exception {
-		String[] command = {RUN_BASH_JAVA_UTILS_CMD_SCRIPT,
-			BashJavaUtils.Command.GET_TM_RESOURCE_DYNAMIC_CONFIGS.toString()};
-		String result = executeScript(command);
+		String result = executeScriptAndFetchLastLine(BashJavaUtils.Command.GET_TM_RESOURCE_DYNAMIC_CONFIGS.toString());
 
 		assertNotNull(result);
```

Comment:
We should assert that the output is prefixed with `BASH_JAVA_UTILS_EXEC_RESULT:`. E2E test feedback is slow.

Reply:
I think the result we get here does not contains the prefix. Do you mean add another test case to test the result from `getTmResourceDynamicConfigs`?

Reply:
Sorry, I forgot that the prefix is removed using `sed` at that point. For a unit test, I think it would be easier to invoke main from BashJavaUtils directly, and redirect System.out using `System.setOut` but that's outside the scope of this PR.

Reply:
Yes indeed. Currently, the test case here is more like an end-to-end test. However, I think it's necessary to test the bash function. What do you think is the best way to cover those code path?

Reply:
Would it be possible to invoke `BashJavaUtils#main` directly, and use `System.setOut` to redirect the output to stdout to a different stream?

Reply:
Sorry, I fear I did not get your point. First, we could directly check the correctness of `BashJavaUtils#main` here. I mean we also need to test `config.sh#runBashJavaUtilsCmd`, verify whether it filters the prefix and matches the regex correctly. After all, current approach looks good to me at the moment.

Reply:
After giving it another thought, I am also fine with the current approach.